buildSrc/src/main/kotlin/org/jetbrains/conventions/kotlin-jvm.gradle.kts

```diff
@@ -0,0 +1,35 @@
+package org.jetbrains.conventions
+
+import org.jetbrains.configureDokkaVersion
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    id("org.jetbrains.conventions.base")
+    kotlin("jvm")
+}
+
+configureDokkaVersion()
+
+val language_version: String by project
+
+tasks.withType<KotlinCompile>().configureEach {
+    kotlinOptions {
+        freeCompilerArgs = freeCompilerArgs + listOf(
+            "-opt-in=kotlin.RequiresOptIn",
+            "-Xjsr305=strict",
+            "-Xskip-metadata-version-check",
+            // need 1.4 support, otherwise there might be problems with Gradle 6.x (it's bundling Kotlin 1.4)
+            "-Xsuppress-version-warnings"
+        )
+        allWarningsAsErrors = true
+        languageVersion = language_version
+        apiVersion = language_version
+        jvmTarget = "1.8"
+    }
+}
+
+// Gradle metadata
+java {
+    withSourcesJar()
```

Comment:
If I understand correctly, `withSourcesJar` adds a task and it's also added as a dependency for `assemble`, so `sources.jar` will be built for projects that have this applied

`kotlin-jvm` is not applied for the gradle plugin and its sources are not published, so it looks like this `withSourcesJar` needs to be a part of the `maven-publish` convention

Reply:
Thanks for checking, and good catches.

I've split out the shared Java config into a new convention plugin, `base-java.gradle.kts`. It sets up the Java Toolchain for all subprojects to be Java 8, and, if the Maven Publish plugin is present, it enables the sources jar.